### ISSUE TITLE
[Bug]: docker-compose up errors during build.

### ISSUE THREAD

**Issue by MicahZoltu** [sd-webui/stable-diffusion-webui#1416]
### What happened?

```
git clone https://github.com/sd-webui/stable-diffusion-webui.git
cd stable-diffusion-webui
docker-compose up --abort-on-container-exit
```

This is on a freshly provisioned cloud host.  Worked fine for me a couple days ago on localhost.

### Version

0.0.1 (Default)

### Where are you running the webui?

Linux

### Custom settings

```shell
# Validate the model files on every container restart
# (useful to set to false after you're sure the model files are already in place)
VALIDATE_MODELS=true

# Automatically relaunch the webui on crashes
WEBUI_RELAUNCH=true

# Which webui to launch
WEBUI_SCRIPT=webui_streamlit.py
# WEBUI_SCRIPT=webui.py

# Pass cli arguments to webui.py e.g:
# WEBUI_ARGS=--optimized --extra-models-cpu --gpu=1 --esrgan-gpu=1 --gfpgan-gpu=1
WEBUI_ARGS=

STREAMLIT_SERVER_HEADLESS=true
```
```


### Relevant log output

```shell
Step 7/19 : RUN pip install -r /install/requirements.txt
 ---> Running in 506180e22c6e
Collecting git+https://github.com/crowsonkb/k-diffusion.git (from -r /install/requirements.txt (line 28))
  Cloning https://github.com/crowsonkb/k-diffusion.git to /tmp/pip-req-build-uxz0rak9
  Running command git clone --filter=blob:none --quiet https://github.com/crowsonkb/k-diffusion.git /tmp/pip-req-build-uxz0rak9
  Resolved https://github.com/crowsonkb/k-diffusion.git to commit 567e11f7062ba20ae32b5a8cd07fb0fc4b9410cf
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Installing backend dependencies: started
  Installing backend dependencies: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Obtaining latent-diffusion from git+https://github.com/devilismyfriend/latent-diffusion#egg=latent-diffusion (from -r /install/requirements.txt (line 76))
  Cloning https://github.com/devilismyfriend/latent-diffusion to ./src/latent-diffusion
  Running command git clone --filter=blob:none --quiet https://github.com/devilismyfriend/latent-diffusion /install/src/latent-diffusion
  Resolved https://github.com/devilismyfriend/latent-diffusion to commit 6d61fc03f15273a457950f2cdc10dddf53ba6809
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
ERROR: Ignored the following versions that require a different python version: 1.22.0 Requires-Python >=3.8; 1.22.0rc1 Requires-Python >=3.8; 1.22.0rc2 Requires-Python >=3.8; 1.22.0rc3 Requires-Python >=3.8; 1.22.1 Requires-Python >=3.8; 1.22.2 Requires-Python >=3.8; 1.22.3 Requires-Python >=3.8; 1.22.4 Requires-Python >=3.8; 1.23.0 Requires-Python >=3.8; 1.23.0rc1 Requires-Python >=3.8; 1.23.0rc2 Requires-Python >=3.8; 1.23.0rc3 Requires-Python >=3.8; 1.23.1 Requires-Python >=3.8; 1.23.2 Requires-Python >=3.8; 1.23.3 Requires-Python >=3.8
ERROR: Could not find a version that satisfies the requirement numpy==1.22.0 (from versions: 1.3.0, 1.4.1, 1.5.0, 1.5.1, 1.6.0, 1.6.1, 1.6.2, 1.7.0, 1.7.1, 1.7.2, 1.8.0, 1.8.1, 1.8.2, 1.9.0, 1.9.1, 1.9.2, 1.9.3, 1.10.0.post2, 1.10.1, 1.10.2, 1.10.4, 1.11.0, 1.11.1, 1.11.2, 1.11.3, 1.12.0, 1.12.1, 1.13.0rc1, 1.13.0rc2, 1.13.0, 1.13.1, 1.13.3, 1.14.0rc1, 1.14.0, 1.14.1, 1.14.2, 1.14.3, 1.14.4, 1.14.5, 1.14.6, 1.15.0rc1, 1.15.0rc2, 1.15.0, 1.15.1, 1.15.2, 1.15.3, 1.15.4, 1.16.0rc1, 1.16.0rc2, 1.16.0, 1.16.1, 1.16.2, 1.16.3, 1.16.4, 1.16.5, 1.16.6, 1.17.0rc1, 1.17.0rc2, 1.17.0, 1.17.1, 1.17.2, 1.17.3, 1.17.4, 1.17.5, 1.18.0rc1, 1.18.0, 1.18.1, 1.18.2, 1.18.3, 1.18.4, 1.18.5, 1.19.0rc1, 1.19.0rc2, 1.19.0, 1.19.1, 1.19.2, 1.19.3, 1.19.4, 1.19.5, 1.20.0rc1, 1.20.0rc2, 1.20.0, 1.20.1, 1.20.2, 1.20.3, 1.21.0rc1, 1.21.0rc2, 1.21.0, 1.21.1, 1.21.2, 1.21.3, 1.21.4, 1.21.5, 1.21.6)
ERROR: No matching distribution found for numpy==1.22.0
The command '/bin/bash -c pip install -r /install/requirements.txt' returned a non-zero code: 1
ERROR: Service 'stable-diffusion' failed to build : Build failed
```
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hlky** [comment]
The dockerfile on the repo is outdated. Now it is intended for docker users to just [use the docker image](https://hub.docker.com/r/hlky/sd-webui)

`docker pull hlky/sd-webui:latest`

The reason is that we now use our [own pytorch base image](https://hub.docker.com/r/hlky/pytorch) with python 3.8, the Dockerfile is not set to use that base image

**Comment by MicahZoltu** [comment]
Is there a good docker-compose to use with that image?  Is it compatible with the volumes from the one in here?

**Comment by hthetiot** [comment]
I had the same issue and applied thoses fixes to be able to build the docker image :

```
diff --git a/requirements.txt b/requirements.txt
index 1c6e6a4..14ea78c 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -3,7 +3,7 @@
 # Minimum Environment Dependencies for Stable Diffusion
 #torch  # already satisfied as 1.12.1 from base image
 #torchvision  # already satisfied as 0.13.1 from base image
-numpy==1.21.5  # already satisfied as 1.21.5 from base image
+numpy==1.22.0  # already satisfied as 1.21.5 from base image


 # Stable Diffusion (see: https://github.com/CompVis/stable-diffusion)
@@ -11,7 +11,6 @@ transformers==4.22.2
 diffusers==0.3.0
 invisible-watermark==0.1.5
 pytorch_lightning==1.7.7
-open_clip_torch==2.0.2

 # Additional Stable Diffusion Requirements
 # TODO: Pin external dependency versions
```

TLDR, set numpy to 1.22.0 and add open_clip_torch==2.0.2 to requirements.txt.

While I understand that using pre-made image is more easy for most users, it is important to me to understand the content of the image that is running on my host by building it myself for security reasons. I also beleive that adding a continious integration to make sure docker image can be built can be usefull for user that want like me to build the image.

**Comment by hlky** [comment]
@hthetiot You're still using a pre-made pytorch image so your concerns about security are moot.

**Comment by hlky** [comment]
[Dockerfile_base](https://github.com/sd-webui/stable-diffusion-webui/blob/dev/Dockerfile_base) can be used to build the base image, this is the pytorch image plus the required dependencies.
`sudo DOCKER_BUILDKIT=1 docker build --progress=auto -t docker.io/hlky/sd-webui:base .`

[Dockerfile](https://github.com/sd-webui/stable-diffusion-webui/blob/dev/Dockerfile) can be used to build the main image, this is the base image plus the webui.

` sudo DOCKER_BUILDKIT=1 docker build --progress=auto --build-args IMAGE=hlky/sd-webui:base -t docker.io/hlky/sd-webui:latest .`

```
docker run --rm -p 8501:8501 -e "VALIDATE_MODELS=true" -v "model_cache:/sd/model_cache/" -v "${PWD}/outputs:/sd/outputs" --gpus all hlky/sd-webui:latest
```

**Comment by hthetiot** [comment]
> @hthetiot You're still using a pre-made pytorch image so your concerns about security are moot.

@hlky Well, I have access to the "pre-made pytorch" image source (that i actualy patch locally for performance), I don't have acess to your pytorch image. So not moot, Sorry.

**Comment by hlky** [comment]
It's built from the same source.

If your own pytorch image is not using python3.8 you will have issues and receive no support for those issues.

**Comment by hthetiot** [comment]
Thank you @hlky for confirming the source of your image is official pytoch. Understood for python 3.8.
I saw your commit on new docker, I may do a PR to restore docker-compose later.

**Comment by MicahZoltu** [comment]
Would you be open to a PR that makes the above change to combine the two Dockerfiles into one Dockerfile, and then updating `docker-compose.yml` to use it?

If so, is there any CI that would need updating for pushing out the base/final image, or is that currently done manually?
